llvm/lib/Target/SPIRV/SPIRVISelLowering.h

```diff
@@ -31,6 +31,10 @@ class SPIRVTargetLowering : public TargetLowering {
     return true;
   }
 
+  // Set the lower limit for number of blocks in a jump table in such a way to
+  // prevent creation of jump tables from switch
+  unsigned getMinimumJumpTableEntries() const override { return 0xffffffff; }
```

Comment:
Wouldn't override `areJTsAllowed` more correct?
https://llvm.org/doxygen/classllvm_1_1TargetLoweringBase.html#a392f6e72d46ff14ee31481e3452f6c31

Reply:
Thank you, I think you are right, `areJTsAllowed` looks better. Changed

Reply:
Curious, what happens if the `-fjump-tables` option is used. Do we ignore it?

Reply:
I think we are forced to manage this on a lower level than user options. This backend clearly doesn't support jump tables at the moment, so lowering switches indeed should ignore -fjump-tables/-fno-jump-tables until the corresponding support is added to the project.

Reply:
Sure, wondering if a warning/error should be emitted in the driver if this flag is used with the spirv target.

Reply:
It's a good question indeed, thank you. Description of the command line options doesn't help to decide, it's just "Use jump tables for lowering switches" that we can interpret in different ways. Probably, the main argument would be that jump tables provide means of optimization (better code) and at maximum this deserves a warning, not an error. As for warnings, I'm not sure if we produce any at the moment, but all that helps a developer to better understand the job done would help of course. I should remember this idea in general and think how we can use it.